### DISCUSSION TITLE
Changing keyboard layout not working

### DISCUSSION THREAD

**Discussion by thejohnbrown** [i3/i3#4900] [Q&A]
Hello,

I am trying to change my keyboard layout from US to GB. I have the following in my i3 config file `exec_always "setxkbmap -layout gb"`, however when I login the keyboard layout is still US. I have tested this command in my terminal and it switches the keyboard layout to GB and also if I reload i3 with mod+shift+r the GB keyboard layout is applied. Does anyone have any idea why this is? Is it to do with the placement of this particular line in my config?

Thanks.

**Comment by Airblader** [comment]
Try redirecting stdout and stderr of the command to a file and see if there are any errors in it after starting i3.

**Comment by thejohnbrown** [comment]
Redirected stdout and stderr to separate files and both are empty.

**Comment by mohamadali-halwani** [comment]
Instead of 
>  `exec_always "setxkbmap -layout gb"`

Try using `exec "setxkbmap -layout gb"` It worked for me like that
